evm/p2p/test_ecies.py

```diff
@@ -0,0 +1,83 @@
+     ),
+]
+
+
+def test_encrypt_decrypt():
+    msg = b'test yeah'
+    privkey = ecies.generate_privkey()
+    ciphertext = ecies.encrypt(msg, privkey.public_key)
+    decrypted = ecies.decrypt(ciphertext, privkey)
+    assert decrypted == msg
+
+    privkey2 = ecies.generate_privkey()
+    with pytest.raises(ecies.DecryptionError):
+        decrypted = ecies.decrypt(ciphertext, privkey2)
+
+
+def test_decrypt_known_good_handshake():
+    # Data taken from https://gist.github.com/fjl/3a78780d17c755d22df2
+    privkey = keys.PrivateKey(
+        decode_hex("c45f950382d542169ea207959ee0220ec1491755abe405cd7498d6b16adb6df8"))
+    auth_ciphertext = decode_hex(
+        "04a0274c5951e32132e7f088c9bdfdc76c9d91f0dc6078e848f8e3361193dbdc43b94351ea3d89e4ff33ddcefbc80070498824857f499656c4f79bbd97b6c51a514251d69fd1785ef8764bd1d262a883f780964cce6a14ff206daf1206aa073a2d35ce2697ebf3514225bef186631b2fd2316a4b7bcdefec8d75a1025ba2c5404a34e7795e1dd4bc01c6113ece07b0df13b69d3ba654a36e35e69ff9d482d88d2f0228e7d96fe11dccbb465a1831c7d4ad3a026924b182fc2bdfe016a6944312021da5cc459713b13b86a686cf34d6fe6615020e4acf26bf0d5b7579ba813e7723eb95b3cef9942f01a58bd61baee7c9bdd438956b426a4ffe238e61746a8c93d5e10680617c82e48d706ac4953f5e1c4c4f7d013c87d34a06626f498f34576dc017fdd3d581e83cfd26cf125b6d2bda1f1d56")  # noqa: E501
+    auth_plaintext = decode_hex(
+        "884c36f7ae6b406637c1f61b2f57e1d2cab813d24c6559aaf843c3f48962f32f46662c066d39669b7b2e3ba14781477417600e7728399278b1b5d801a519aa570034fdb5419558137e0d44cd13d319afe5629eeccb47fd9dfe55cc6089426e46cc762dd8a0636e07a54b31169eba0c7a20a1ac1ef68596f1f283b5c676bae4064abfcce24799d09f67e392632d3ffdc12e3d6430dcb0ea19c318343ffa7aae74d4cd26fecb93657d1cd9e9eaf4f8be720b56dd1d39f190c4e1c6b7ec66f077bb1100")  # noqa: E501
+    decrypted = ecies.decrypt(auth_ciphertext, privkey)
+    assert auth_plaintext == decrypted
+
+
+@pytest.mark.parametrize("privkey_hex, pubkey_hex, ecdh_expected", STATIC_ECDH_VALUES)
+def test_ecdh(privkey_hex, pubkey_hex, ecdh_expected):
+    privkey = keys.PrivateKey(decode_hex(privkey_hex))
+    pubkey = keys.PublicKey(decode_hex(pubkey_hex))
+    assert ecdh_expected == encode_hex(ecies.ecdh_agree(privkey, pubkey))
+
+
+# FIXME: Document those values; this test was lifted from pydevp2p
+def test_hmac_sha256():
+    k_mac = decode_hex("0x07a4b6dfa06369a570f2dcba2f11a18f")
+    indata = decode_hex("0x4dcb92ed4fc67fe86832")
+    hmac_expected = decode_hex("0xc90b62b1a673b47df8e395e671a68bfa68070d6e2ef039598bb829398b89b9a9")
+    hmac = ecies.hmac_sha256(k_mac, indata)
+    assert hmac_expected == hmac
+
+    # message tag generated by geth
+    tag_secret = decode_hex("0xaf6623e52208c596e17c72cea6f1cb09")
+    tag_input = decode_hex("0x3461282bcedace970df2")
+    tag_expected = decode_hex("0xb3ce623bce08d5793677ba9441b22bb34d3e8a7de964206d26589df3e8eb5183")
+    hmac = ecies.hmac_sha256(tag_secret, tag_input)
+    assert hmac == tag_expected
+
+
+# FIXME: Document those values; this test was lifted from pydevp2p
```

Comment:
Couple of remaining `FIXME` left in.  Fine to fix them later but it might be good to at least provide a link here for now.

Reply:
Included the links in the comments